src/main/java/edu/harvard/iq/dataverse/util/MarkupChecker.java

```diff
@@ -11,56 +6,105 @@
 import org.jsoup.parser.Parser;
 
 /**
- * Wrapper for Jsoup clean
- * 
+ * Provides utility methods for sanitizing and processing HTML content.
+ * <p>
+ * This class serves as a wrapper for the {@code Jsoup.clean} method and offers
+ * multiple configurations for cleaning HTML input. It also provides a method
+ * for escaping HTML entities and stripping all HTML tags.
+ * </p>
+ *
  * @author rmp553
  */
 public class MarkupChecker {
-    
-    
-    
+
     /**
-     * Wrapper around Jsoup clean method with the basic Safe list
-     *   http://jsoup.org/cookbook/cleaning-html/safelist-sanitizer
-     * @param unsafe
-     * @return 
+     * Sanitizes the provided HTML content using a customizable configuration.
+     * <p>
+     * This method uses the {@code Jsoup.clean} method with a configurable {@code Safelist}.
+     * For more details, see the
+     * <a href="http://jsoup.org/cookbook/cleaning-html/safelist-sanitizer">Jsoup SafeList Sanitizer</a>.
+     * </p>
+     * <p>
+     * It supports preserving class attributes and optionally adding "noopener noreferrer nofollow"
+     * attributes to anchor tags to enhance security and usability.
+     * </p>
+     *
+     * @param unsafe                    the HTML content to be sanitized; may contain unsafe or untrusted elements.
+     * @param keepClasses               whether to preserve class attributes in the sanitized HTML.
+     * @param includeNoopenerNoreferrer whether to add "noopener noreferrer nofollow" to <a> tags.
+     * @return a sanitized HTML string, free from potentially harmful content.
      */
-    public static String sanitizeBasicHTML(String unsafe) {
-
+    private static String sanitizeHTML(String unsafe, boolean keepClasses, boolean includeNoopenerNoreferrer) {
         if (unsafe == null) {
             return null;
         }
-        // basic includes: a, b, blockquote, br, cite, code, dd, dl, dt, em, i, li, ol, p, pre, q, small, span, strike, strong, sub, sup, u, ul
-        //Whitelist wl = Whitelist.basic().addTags("img", "h1", "h2", "h3", "kbd", "hr", "s", "del");  
 
-        Safelist sl = Safelist.basicWithImages().addTags("h1", "h2", "h3", "kbd", "hr", "s", "del", "map", "area").addAttributes("img", "usemap")
-                .addAttributes("map", "name").addAttributes("area", "shape", "coords", "href", "title", "alt")
+        // Create a base Safelist configuration
+        Safelist sl = Safelist.basicWithImages()
+                .addTags("h1", "h2", "h3", "kbd", "hr", "s", "del", "map", "area")
+                .addAttributes("img", "usemap")
```

Comment:
I'm a little confused what's going on here. Are we allowing more HTML tags and attributes? Does this affect other parts of the app, besides features items? 🤔

Reply:
No. We only allow more HTML tags for featured items, where for example we need to prevent HTML sanitization from removing important tags like className. Previous use cases work as before. The many changes you see are simply due to the refactoring I did to selectively support the new logic for featured items.

Reply:
Ok, thanks.